### DISCUSSION TITLE
SPARKNLP application in R

### DISCUSSION THREAD

**Discussion by ghost** [JohnSnowLabs/spark-nlp#12888] [Q&A] [answered]
Hi, 
I'm reading the paper  about implementing NER in SPARK NLP for Clinical Research, 'Improving Clinical Document Understanding on COVID-19 Research with SPARK NLP'. There are examples of how to implement the NLP pipeline at the back of the paper. I'm using R so following David Kincade's SPARK NLP 
 R implementations at [https://github.com/r-spark/sparknlp/commit/156aea018c37f4fb082399f2d5ee5810c199ade6](url)

What I can't figure out is how to use one of the pretrained models e.g. ner_diseases within this framework. 

I presume it's the following block of code that implements this but not sure where to load model.
Do we need to preload language models like spacy?  Any guidance? Has anyone else implemented these pipelines in R?

```
clinical_ner = NERDLModel.pretrained("
      ner_clinical", "en", "clinical/models")
      \
      setInputCols(["sentence", "token", "
      embeddings"])\
      .setOutputCol("ner")
```

**Comment by maziyarpanahi** [comment, marked as answer]
Hi,

Anything that says `clinical/models` is a licensed model/pipeline and they also have their own annotators which only work with licensed keys. You can ask for trial keys or an academic license. 

In any case, this repository is only for public/open-source products/models. If you have more questions you can ask in Slack under #healthcare channel, they answer pretty quickly.

**Reply by ghost** [reply]
Thank you
